server/services/signup/hooks/getSignupAndEvent.js

```diff
@@ -1,42 +1,69 @@
+const _ = require('lodash');
 const config = require('../../../../config/ilmomasiina.config'); // eslint-disable-line
 const md5 = require('md5');
 
 module.exports = () => (hook) => {
   const models = hook.app.get('models');
   const id = hook.id;
   const editToken = hook.params.query.editToken;
+  const fields = [];
+  const userAnswers = [];
 
   if (editToken !== md5(`${`${hook.id}`}${config.editTokenSalt}`)) {
     throw new Error('Invalid editToken');
   }
 
-  return models.signup
-    .findOne({
-      where: {
-        id,
-      },
-    })
-    .then(signup =>
-      models.quota
-        .findOne({
-          where: {
-            id: signup.quotaId,
-          },
-        })
-        .then(quota =>
-          models.event
-            .findOne({
-              where: {
-                id: quota.eventId,
-              },
-            })
-            .then((event) => {
-              hook.result = {
-                signup,
-                event,
-              };
-              return hook;
-            }),
-        ),
+  return models.answer
+    .findAll({ where: { signupId: id } })
+    .then(answers => answers.map(answer => userAnswers.push(answer.dataValues)))
+    .then(() => models.signup
+      .findOne({
+        where: {
+          id,
+        },
+      })
+      .then(signup => {
+        if (signup === null) { // Event not found with id, probably deleted
+          hook.result = {
+            signup,
+            event: null,
+          };
+          return hook;
```

Comment:
The original code would fail if the deletion link was clicked after the deletion, so instead of just crashing silently and returning 500 to client we'll send an "empty" response